bigquery/google/cloud/bigquery/dbapi/cursor.py

```diff
@@ -153,8 +153,8 @@ def execute(self, operation, parameters=None, job_id=None):
         # Wait for the query to finish.
         try:
             self._query_job.result()
-        except google.cloud.exceptions.GoogleCloudError:
-            raise exceptions.DatabaseError(self._query_job.errors)
+        except google.cloud.exceptions.GoogleCloudError as e:
+            raise exceptions.DatabaseError(self._query_job.errors or e)
```

Comment:
This change makes the type of the exception's `args` attribute ambiguous:  it will be either a list of mappings or a single exception.  I don't believe we should force users to guess here.

Reply:
@tseaver my main goal was to highlight the issue.

It also appears that `execptions.DatabaseError` doesn't document getting unconventional argument (the common assumption with Python exception is that they receive a string, in this case it appears that `self._query_job.errors` a list of dicts). I know that the DBAPI PEP (249) isn't clear about the values expected here (it says: `The values of these exceptions are not defined. They should give the user a fairly good idea of what went wrong, though.`).

If a list of dicts is expected, we should document it here and enforce that consistency:
https://github.com/googleapis/google-cloud-python/blob/0e2c54ce54909467b202a73f04fc3aa4a29a749b/bigquery/google/cloud/bigquery/dbapi/exceptions.py#L30

Reply:
For context, I'm working on integrating BigQuery with Superset. Superset integrates with most SQL-speaking database out there. You'll understand that we like when DBAPIs behave in a consistent way :)
https://github.com/apache/incubator-superset

Reply:
I think we could probably forget `self._query_job.errors` entirely, as any error information should be present in the exception. I don't remember the reason I used the `errors` from the job resource to begin with.

Reply:
@tswast Any issues about backward-compatibility for folks who might be catching the `DatabaseError` if we just propagate the `GoogleCloudError` instead?

Reply:
We do need to catch and throw as `DatabaseError` per PEP 249. https://www.python.org/dev/peps/pep-0249/#databaseerror